### PR TITLE
added Corp Type Module

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* /bcgov/entity#6393

**NB: I have tested this in-situ in Filings UI (and will be importing it there shortly).**

*Description of changes:*
- added Corp Type Module
- added unit test file

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

File: src/modules/corp-type-module/corp-type-module.ts [hunk 1]
```diff
@@ -0,0 +1,396 @@
...
+  CONT_IN_SOCIETY = 'CS',
+  ULC_CONTINUE_IN = 'CUL',
+  EXTRA_PRO_REG = 'EPR',
+  FINANCIAL = 'FI',
+  FOREIGN = 'FOR',
+  PARTNERSHIP = 'GP',
+  LIBRARY = 'LIB',
+  LICENSED = 'LIC',
+  LL_PARTNERSHIP = 'LL',
+  LIMITED_CO = 'LLC',
```
Comment (cameron-freshworks) [inline src/modules/corp-type-module/corp-type-module.ts:29]: Just curious what these are mapped from. 
In Lear a limited company is 'BC'. 
Also, in Name Request an 'LLC' is a Limited `Liability` Company'

Seems challenging to keep things in sync! Are `corpType`'s mapping back to something in Auth?

Comment (cameron-freshworks) [reply src/modules/corp-type-module/corp-type-module.ts:29]: Maybe i just took the abbreviated name too literally, as it is call `LIMITED_CO`. So maybe i'm mistaken here..

**Review by cameron-freshworks — APPROVED** [review]
Looks good, just a minor question!

File: src/modules/corp-type-module/corp-type-module.ts (hunk 1 above)
Comment (severinbeauvais) [reply src/modules/corp-type-module/corp-type-module.ts:29]: This is from a spreadsheet that Linda attached to the ticket.

For these key names I used the "short desc" as that is sort-of meaningful.

There is another list, "entity types", which is used by NRs, which contains some other types. Combining these will require another ticket... and maybe getting the BAs to take a crack at it first.

Specifically, here, don't worry about "LMITED_CO = LLC". The first is an internal key name and the second is the actual corp type cd in the dbs.

**Pull request merged**